### PR TITLE
[13.0] use algolia v2 python client

### PR DESCRIPTION
**Pull request by PierrickBrun**

### CODE REVIEW

**Comment by simahawk** [comment]
@PierrickBrun from customer proj perspective: is anything else needed to integrate this?  (beside pinning the new client version)

**Comment by PierrickBrun** [comment]
@simahawk I think everything is done here but this needs to be tested on a real installation

**Comment by simahawk** [comment]
> @simahawk I think everything is done here but this needs to be tested on a real installation

I see. Let me know when is tested on your side. Then I can plan testing in our project :pray:

**Comment by PierrickBrun** [comment]
> > @simahawk I think everything is done here but this needs to be tested on a real installation
> 
> I see. Let me know when is tested on your side. Then I can plan testing in our project pray

Hi @simahawk sorry I did not follow up on this. I don't have a project using algolia that I could use to test this, could you do it on your side ?

File: setup/connector_algolia/setup.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
     odoo_addon={
         'external_dependencies_override': {
             'python': {
-                'algoliasearch': 'algoliasearch>=1.0.0,<2.0.0',
+                'algoliasearch': 'algoliasearch>=2.0,<3.0',
```
Comment (sbidoul) [inline setup/connector_algolia/setup.py]: In v13+ the version specifier can be declared in the manifest external dependencies.

**Comment by hparfr** [comment]
@simahawk can you finish your review on this one ?

File: setup/connector_algolia/setup.py (hunk 1 above)
Comment (PierrickBrun) [reply setup/connector_algolia/setup.py]: done

**Comment by simahawk** [comment]
I don't have time to test now: are we sure this works on existing instances?

**Comment by sbidoul** [comment]
Note that in Odoo 13+ you can specify the versions in the manifest external dependencies.

**Comment by PierrickBrun** [comment]
> Note that in Odoo 13+ you can specify the versions in the manifest external dependencies.

It's done like this here (If I'm not mistaken)

File: test-requirements.txt [hunk 2]
```diff
@@ -1,6 +1,6 @@
 odoo-test-helper>=1.1.0
 
-algoliasearch<2.0.0
+algoliasearch>=2.0,<3.0
 elasticsearch>=7.0.0,<8.0.0
```
Comment (sbidoul) [inline test-requirements.txt]:
```suggestion
```

These two are redundant with requirements.txt

**Comment by sbidoul** [comment]
@PierrickBrun sorry, un bug dans la matrice :)

**Review by sebastienbeau — APPROVED** [review]
LGTM code review

**Comment by simahawk** [comment]
@PierrickBrun finally had time to test w/ real data and real Algolia. Seems to work, tnx!

**Pull request merged**